packages/zoe/contracts/autoswap.js

```diff
@@ -21,303 +19,190 @@ export const makeContract = harden((zoe, terms) => {
   const liquidityAssay = liquidityMint.getAssay();
   const assays = [...startingAssays, liquidityAssay];
 
-  let poolOfferHandle;
+  const LIQ_INDEX = 2;
+  let poolHandle;
   let liqTokenSupply = 0;
 
-  const { add, subtract, multiply, divide } = natSafeMath;
+  const { subtract, multiply, divide } = natSafeMath;
 
-  const addLiquidity = async escrowReceipt => {
-    const extentOpsArray = zoe.getExtentOpsArray();
+  return zoe.addAssays(assays).then(() => {
+    const unitOpsArray = zoe.getUnitOpsForAssays(assays);
     const {
```

Comment:
Should verify that all these `unitOps` use the `natExtentOps`. If not, we should fail early. Probably here is the earliest.

Reply:
Good idea. I added that check